File: interfaces/samples/delphi/hello/hello.pas
unit Hello;

interface

uses
  SysUtils, WinTypes, WinProcs, Messages, Classes, Graphics, Controls,
  Forms, Dialogs, StdCtrls, Amzi;

type
  TForm1 = class(TForm)
    Hello: TButton;
    ls: TLSEngine;

    procedure HelloClick(Sender: TObject);
  private
  public
    { Public declarations }
  end;

var
  Form1: TForm1;

implementation

{$R *.DFM}

procedure TForm1.HelloClick(Sender: TObject);
var
  t: TTerm;
  s: string;
begin
  ls.InitLS('hello');  { Initialize engine }
  ls.LoadXPL('hello'); { Load .xpl file }
  { Query the Logic Server }
  ls.ExecPStr(t, 'hello($Delphi Programmer$, X)');
  { Get the second argument of the result back }
  s := ls.GetPStrArg(t, 2);
  ShowMessage(s);  { Display the result }
  ls.Close;  { Close engine }
end;

end.

File: samples/java/hello/Hello.java
import amzi.ls.*;

class Hello {
  public static void main(String args[]) throws LSException {
    long term;
    String result;

    LogicServer ls = new LogicServer();
    ls.Init("");
    ls.Load("hello.xpl");
    term = ls.ExecStr("hello($Java Programmer$, Response)");
    if (term == 0)
	  System.out.println("Hello Failed");
	else
      {
	  result = ls.GetStrArg(term, 2);
      System.out.println(result);
	  }
    ls.Close();
  }
}
